nanoFirmwareFlasher.Library/FileDeployment/DeploymentFile.cs

```diff
@@ -0,0 +1,15 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace nanoFramework.Tools.FirmwareFlasher.FileDeployment
+{
+    internal class DeploymentFile
+    {
+        public string FileName { get; set; }
```

Comment:
I suggest these names make it absolutely clear what's their purpose!
Something like: StoragePath and SourcePath or StorageFile and SourceFile , or....

Reply:
What's not clear with FileName and ContentFileName?
In C#, FileName property represents a full path qualified name (like the dialog box). It can in some circonstances not include the full path (email attachments in Mime or in the Web Header when you download a file but that's not .NET).
Path is ambiguous as well; it does not always refer to a fully qualified path including the name of the file.
So, the framework does not necessarily helps us :-)
So, what about:

* DestinationFilePath
* SourceFilePath

Reply:
I agree with those suggestion above. The names make very clear which one is which!